homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT first_name, last_name, salary
+from employees
+WHERE salary < (
+    SELECT salary
+    FROM employees
+    WHERE employee_id = 182
+    );
+
 -- 6. the first name of all employees including the first name of their manager.
+SELECT e1.first_name, e2.first_name AS manager_name
+FROM employees e1, employees e2
+WHERE e2.employee_ID = e1.manager_ID;
+
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT e1.first_name, e2.first_name AS manager_name
+FROM employees e1
+    LEFT JOIN employees e2 ON e1.employee_id = e2.manager_id;
+
 -- 8. the details of employees who manage a department.
+SELECT *
+FROM employees
+WHERE employee_id IN (
+    SELECT manager_id
+    FROM departments D2
+    );
+
 -- 9. the first name, last name, and department number for those employees who works in the same department as the employee who holds the last name as Taylor.
+SELECT first_name, last_name, department_id
+FROM employees
+WHERE department_id IN (
+    SELECT DEPARTMENT_ID
+    FROM employees
+    WHERE last_name = 'Taylor'
+    );
+
 --10. the department name and number of employees in each of the department.
---11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT department_name, COUNT(*) AS employess_count
+FROM employees
+    INNER JOIN departments USING (department_id)
+GROUP BY department_name;
+
+--11.1 the name of the department, average salary and number of employees working in that department who got commission.
+--(display departments even if there are no workers with commission)
+SELECT department_name, AVG(salary), COUNT(salary) AS employee_count
+FROM departments
+    LEFT JOIN (
+        SELECT salary, department_id
+        FROM employees
+        WHERE commission_pct IS NOT NULL
+        ) USING (department_id)
+GROUP BY department_name;
+
+--11.2 the name of the department, average salary and number of employees working in that department who got commission.
+--(do NOT display departments if there are no workers with commission)
+SELECT department_name, AVG(salary), COUNT(salary) AS employee_count
+FROM employees
+         INNER JOIN departments USING (department_id)
+WHERE commission_pct IS NOT NULL
+GROUP BY department_name;
+
 --12. job title and average salary of employees.
+SELECT job_title, AVG(salary) AS everage_salary
+FROM jobs
+    INNER JOIN employees USING (job_id)
+GROUP BY job_title;
+
 --13. the country name, city, and number of those departments where at least 2 employees are working.
+SELECT country_name, city, COUNT(employee_id) AS employees_count
+FROM locations
+    INNER JOIN countries USING (country_id)
+    INNER JOIN departments USING (location_id)
+    INNER JOIN employees USING (department_id)
+GROUP BY country_name, city
+HAVING COUNT(employee_id) >= 2;
```

Comment:
Why we can't use **department_id** instead of **COUNT(employee_id)** and in **GROUP BY specify country_name, city, department_id**, since from what I understood, "NUMBER OF THOSE DEPARTMENTS" is meant as the ID of those departments.

Reply:
Refactored to use **department_id** instead of **COUNT(employee_id)**